renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -68,6 +139,103 @@ void VkAccelerationStructureInfo::Release()
     delete this;
 }
 
+uint64_t VkAccelerationStructureInfo::GetSerialisedSize() const
+{
+  uint64_t geomDataSize = 0;
+  for(const GeometryData &geoData : geometryData)
+    geomDataSize += geoData.GetSerialisedSize();
+
+  const uint64_t size = sizeof(VkAccelerationStructureTypeKHR) +          // type
+                        sizeof(VkBuildAccelerationStructureFlagsKHR) +    // flags
+                        sizeof(uint64_t) + geomDataSize;                  // geometryData;
+
+  return size;
+}
+
+rdcarray<VkAccelerationStructureGeometryKHR> VkAccelerationStructureInfo::convertGeometryData() const
+{
+  rdcarray<VkAccelerationStructureGeometryKHR> result;
+  result.reserve(geometryData.size());
+
+  for(const VkAccelerationStructureInfo::GeometryData &g : geometryData)
+  {
+    VkAccelerationStructureGeometryDataKHR geoUnion = {};
+    switch(g.geometryType)
+    {
+      case VK_GEOMETRY_TYPE_TRIANGLES_KHR:
+      {
+        VkDeviceOrHostAddressConstKHR vData;
+        vData.deviceAddress = 0x0;
+
+        VkDeviceOrHostAddressConstKHR iData;
+        iData.deviceAddress = 0x0;
+
+        // vkGetAccelerationStructureBuildSizesKHR just checks if the transform BDA is non-null,
+        // so fudge that here
+        VkDeviceOrHostAddressConstKHR tData;
+        tData.deviceAddress = g.tris.hasTransformData ? 0x1 : 0x0;
+
+        geoUnion.triangles = VkAccelerationStructureGeometryTrianglesDataKHR{
```

Comment:
Please do not use this syntax. The typename can be omitted here if you're initialising POD with `{}` but either use a normal constructor or a POD brace list.

Reply:
This line is still using the odd construction syntax.